openkfw/innoverse
# Database redesign

**Changes in this PR**:
- [x] updated data model to combine all Comment tables (Comment, NewsComment, PostComment, ProjectComment, CollaborationComment) into a single table **Comment**
- [x] migrate data from old tables into new one (in the [migration file](https://github.com/openkfw/innoverse/pull/136/files#diff-443d5853046a2b72963cf92b8a8f5960b57412016339a3121d63fea18501bf88))
- [x] update all types, requests and queries
- [x] add a separate CommentLike table (like for comments)
- [x] update the old Like comment (rename projectId -> objectId, add objectType) -> for now we only have project likes but we could extend it


**How to test:**
- build all containers in main (build db from scratch e.g. `docker compose up --build`)
- make sure your strapi has data
- add some data to the database (either manually or via a postgres dump)
  - `psql "postgresql://inno:hub@localhost:5432/innohub" < ...path_to_dump.sql`
  - connect to psql via `psql "postgresql://inno:hub@localhost:5432/innohub"`
  - look & check the old table format
- do `npm run prisma migrate deploy` in the new branch (this migration should be applied **20241217173509_comment_model**)
- the result should be something like this:
<img width="471" alt="image" src="https://github.com/user-attachments/assets/666a3fce-6c13-4f40-be82-867f2c8e070d" />

- make sure you update the cache by removing the "sync" entry from the redis UI, and then triggering the` /api/redis/full-refresh` endpoint


other useful psql commands:
- to drop everything from the DB `drop schema innohub CASCADE;`
- to set the "innohub" schema as the current search path in psql -> `SET search_path TO innohub;`

Closes #135
